### ISSUE TITLE
Touchscreen fix

### ISSUE THREAD

**Issue by ryanrudolfoba** [jlobue10/SteamDeck_rEFInd#31]
Hi there. I hope you dont mind as i saw some users are asking for touchscreen support.

Its an easy fix and documented on the rEFInd config - you cant use enable touch and enable mouse at the same time. You only have to choose 1. If you choose both, then the last entry takes precedence and the other gets disabled.

I saw in your config that you defined enable touch followed by enable mouse. So it process enable mouse instead and disables enable touch.

Here is sample of the config i use. I hope it helps.

![Screenshot_20230125-120459](https://user-images.githubusercontent.com/98122529/214633564-41850577-579f-407f-bb39-6a3cdcbd7587.png)

**Comment by jlobue10** [comment]
Thank you! This is good feedback. Testing this now.

**Comment by jlobue10** [comment]
Valve needs a firmware and/ or a driver fix to properly support rEFInd's touch screen option (assuming rEFInd's touch option is not the problem).

**Comment by jlobue10** [comment]
It's possible for us to get this working with drivers added for rEFInd. Needs more investigation. I will happily merge a fix if someone else figures it out (after testing it myself of course).

**Comment by ryanrudolfoba** [comment]
No worries. For now Im ok with just scripting everything no GUI.

**Comment by jlobue10** [comment]
Yep, no worries. This was good feedback, as I was not aware that these two options would only load the last one enabled (and supersede the previous setting. Thank you for that information.

And yes, the config file for Clover seems a bit more daunting, but the framework is there in my code as a starting point for you to do something similar with Clover with a GUI. I did most of the development work through a VNC session on a Kubuntu computer. You can download Qt Creator for free for many Linux distributions, through the Discover app if you're using KDE Plasma. When I was finally able to test, I moved the files over to the Steam Deck, installed Qt Creator on my Steam Deck and then figured out the nuances there. Sudo commands would not function properly from the development version of the GUI, just as an FYI, but I knew that they worked from my previous testing. The install-GUI script also installs the minimum dependencies in order to compile, with `hwinfo` as well, which is needed for sudo-less grabbing of partition GUID info. Once the program was compiled though, it worked as I expected. That's just something to be ware of, if you go down the GUI route at some point. Thanks again for the feedback and info.

**Comment by ryanrudolfoba** [comment]
Yes the Clover config is overwhelming - 1100+ lines. But there is already a configuration GUI for it to easily edit it, but i dont wanna add it yet to the README as it might cause more issues in the end. It just needs python which is already installed by default in SteamOS so no extra dependency is needed.

Other than that i think im ready for the next release - i got some pretty good icons from the community which i will bundle in.

**Comment by ryanrudolfoba** [comment]
Closing this issue. Config setting provided but apparently this might also need a 3rd party driver to fix completely.

**Comment by jlobue10** [comment]
Yes, thank you for the feedback. I will continue to investigate if a generic EFI touchscreen driver can be found (and works) to fix this issue.

Interesting about Clover, but I hear Mac or Mac related, and I'm immediately not interested... Why do they make simple things not simple?!? But I digress. Thanks again.

**Comment by ryanrudolfoba** [comment]
Just another tidbit about Clover - yes generally it is used for hackintosh. You can also load custom ACPI / DSDT tables. I've already disassembled my DSDT and try to patch it for Fan Control but this is all foreign to me so hopefully someone else can step in and conti
